### DISCUSSION TITLE
Migrating from express to fastify

### DISCUSSION THREAD

**Discussion by ghost** [fastify/fastify#3450] [Q&A]
### In express i'm using the following

*middleware.js*
```
export default callback => async (req, res) => {
    try {
        const result = await callback(req.params, req.body)
        if(result) {
            res.send(result)
        }
    } catch(e) {
        res.end()
    }
}
```

*server.js*
```
express.Router.get('/', middlewareHandler(async (params, body) => ({params, body}) ))
```
...so basically if something is returned from the expressRouter, it is output on the page.

### I've tried implementing something similar in fastify, but this is as far as i've come
*middleware.js*
```
export default callback => async (req, rep) => {
    try {
        return { params: req.params, body: req.body }
    } catch(e) {
        res.hijack()
    }
}
```

*server.js*
```
fastify.get('/', async (req, rep) => {
    const response = await middlewareHandler(req, rep)
    if(result) {
        res.send(result)
    }
)
```
...so basically i have to get the result, and then manually get the value and send the value to the page

#### Is there any way I can use the first one (the one was using in express), or something similar, in fastify?

**Comment by ghost** [comment]
Nevermind, I figured it out

**Reply by jsumners** [reply]
This is not an answer.
